_posts/2020-02-24-qute.adoc

```diff
@@ -0,0 +1,182 @@
+
+* The https://quarkus.io/guides/qute-reference#syntax-and-building-blocks[syntax] is mainly inspired by https://handlebarsjs.com/[Handlebars] and https://www.dustjs.com/[Dust.js].
+* The https://quarkus.io/guides/qute-reference#include_helper[template inheritance] is inspired by https://en.wikipedia.org/wiki/Facelets[Facelets] and https://docs.djangoproject.com/en/3.0/ref/templates/language/[Django]. 
+* Qute supports the https://en.wikipedia.org/wiki/Elvis_operator[elvis operator] you might be familiar with from https://kotlinlang.org/[Kotlin]. 
+* https://quarkus.io/guides/qute-reference#template_extension_methods[Extension methods]  that can be used to extend the data classes with new functionality are also inspired by modern languages.
+* If you come from the world of JSP/JSF/Facelets you'll appreciate that `@Named` CDI beans can be referenced directly in any template through the `inject` namespace, e.g. `{inject:foo.price}`. See https://quarkus.io/guides/qute-reference#injecting-beans-directly-in-templates[Injecting Beans Directly In Templates] for more information.
+
+But that's not all.
+We do introduce new features based on Quarkus principles.  
+
+=== Asynchronous Data Resolution - On The Way To Reactive
+
+When we started to design Qute we had one important aspect in mind - the data resolution API should be asynchronous.
+This allows for better resource utilization and fits the Quarkus reactive story.
+Another useful consequence is that it's possible to use a non-blocking client directly from a template to fetch data from external systems:
+
+.Non-blocking Client Data Fetching Example
+[source,html]
+----
+{@org.acme.Client client} <1>
+<html>
+<body>
+    <h1>Quarkus Open Pull Requests</h1>
+    {#for pull in client.pullRequests} <2>
+        <p>{pull.title} - {pull.user.login}</p>
+    {/for} 
+</body>
+</html>
+----
+<1> Parameter declaration - maps `client` to `org.acme.Client`. See the <<type-safe-templates,next section>> for more information.
+<2> `org.acme.Client#getPullRequests()` is using a non-blocking Vert.x client to fetch the data directly from the GitHub API. Since the data resolution is asynchronous the thread is not blocked and can continue performing some other tasks.
+
+[[type-safe-templates]]
+=== Type-safe Templates
+
+Most of the templating engines are not type-safe, ie. do not prevent type errors.
+It's quite natural because dynamicity in templates is very often practical.
+On the other hand, a user is not protected from tedious errors caused by typos.
+Qute templates can be _optionally_ type-safe.
+What does it actually mean?
+A template may contain one or more *parameter declarations*.
+A parameter declaration binds a concrete type information to a given identifier in the current context.
+And what are the advantages of having a type-safe template?
+
+* Quarkus attempts to _validate all expressions_ that reference a parameter declaration. If an invalid/incorrect expression is found the build fails. 
```

Comment:
You attempt or you do ? This sentence is stress inducing.

Reply:
I'll change the wording to "Quarkus validates ...". My point is that it's still an experimental feature and it's more like "best effort"...